frontend/tests/functions/Formatting.test.tsx

```diff
@@ -57,36 +59,77 @@ describe("extractAndReplaceKeys", () => {
 });
 
 describe("convertingSimulationToChartData", () => {
-    const simulationResult: SimulationResults[] = [
-        {
-            panels: [],
-            modelInput: { concentrations: { H2O: 0.3 }, parameters: { Temperature: 300 }, modelId: "model1" },
-            finalConcentrations: { CO: 0.4, H2O: 0.4 },
-        },
-        {
-            panels: [],
-            modelInput: { concentrations: { H2O: 0.2, COS: 0.5 }, parameters: { Temperature: 300 }, modelId: "model2" },
-            finalConcentrations: { CO: 0.5, H2O: 0.3 },
-        },
-    ];
-
-    it("should have necessary properties in the returning object", () => {
-        const res = convertSimulationsToChartData(simulationResult, ["Exp1", "Exp2"]);
-
-        expect(res.every((item) => Object.prototype.hasOwnProperty.call(item, "label"))).toBe(true);
-        expect(res.every((item) => Object.prototype.hasOwnProperty.call(item, "data"))).toBe(true);
-    });
+    const simulationResults: Record<string, SimulationResults[]> = {
+        Exp1: [
+            {
+                panels: [],
+                modelInput: { concentrations: { H2O: 0.3 }, parameters: { Temperature: 300 }, modelId: "model1" },
+                finalConcentrations: { CO: 0.4, H2O: 0.4 },
+            },
+        ],
+        Exp2: [
+            {
+                panels: [],
+                modelInput: {
+                    concentrations: { H2O: 0.2, COS: 0.5 },
+                    parameters: { Temperature: 300 },
+                    modelId: "model2",
+                },
+                finalConcentrations: { CO: 0.5, H2O: 0.3 },
+            },
+        ],
+    };
 
     it("should have correct data", () => {
-        simulationResult.forEach((simRes) => {
-            const res = convertSimulationsToChartData([simRes], ["Exp1", "Exp2"]);
-
-            res.forEach((dataset) => {
-                dataset.data.forEach((point: { x: string; y: number | null }) => {
-                    expect(Object.keys(simRes.finalConcentrations)).toContain(point.x);
-                    expect(Object.values(simRes.finalConcentrations)).toContain(point.y);
-                });
+        const res = convertSimulationsToChartData(simulationResults);
+
+        res.forEach((dataset) => {
+            dataset.data.forEach((point: { x: string; y: number | null }) => {
+                const [modelId, experimentName] = dataset.label.split(" - ");
+                const experimentSimulations = simulationResults[experimentName];
+
+                if (experimentSimulations) {
+                    const matchingSimulation = experimentSimulations.find((sim) => sim.modelInput.modelId === modelId);
+
+                    if (matchingSimulation) {
```

Comment:
why would this not be matching anything? May worry is that this conversion can be all wrong, give bad results, but this test will be all happy, cause it doesn't have anything to fail on.. It is better to have have "input" and an "expected output" explicitly defined.

or is there something I'm missing?